### PR TITLE
Url redirect filter #33

### PR DESCRIPTION
**Pull request by Ericthilen**
Implements a RedirectFilter that intercepts incoming requests and returns HTTP redirects based on a configurable list of redirect rules.

### Requirements covered
- Configurable redirect rules: sourcePath (regex / wildcard compiled to Pattern), targetUrl, statusCode (301/302)
- Rules evaluated in order (first match wins)
- Stops the pipeline on match
- Sets correct `Location` header + status code
- Logs each redirect for debugging

### Tests
- 301 redirect
- 302 redirect
- No matching rule continues pipeline
- Wildcard matching via RedirectRulesLoader.compileSourcePattern("/docs/*")

### How to test
Run RedirectFilterTest.

Issue #33 

Co-authored by @viktorlindell12 

## Summary by CodeRabbit

* **New Features**
  * Added HTTP request redirection capabilities with support for wildcard and regex-based redirect patterns.
  * Configurable redirect status codes (301 and 302) for flexible redirect behavior.

* **Tests**
  * Added comprehensive unit tests covering redirect functionality, pattern matching, and edge cases.

### CODE REVIEW

**Review by gitnes94 — APPROVED** [review]
Allt ser bra ut nu, jätteduktigt jobbat!

File: src/main/java/org/example/server/FilterChain.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package org.example.server;
+
+import java.util.Objects;
+
+public final class FilterChain {
+
+    private final HttpFilter[] filters;
+    private final TerminalHandler terminal;
+    private int index = 0;
+
```
Comment (gvaguirres) [inline src/main/java/org/example/server/FilterChain.java]: Jag har precis kollat på PR "define and create filter interface" och där finns det en interface som heter FilterChain. Ska vi ha en interface där och en final class med samma namn här som inte implementerar interfacen? Jag tror att det kommer att skapa en konflikt?

**Comment by gvaguirres** [comment]
Jag föreslår att vi pausar merge av specifika filter då jag har märkt att vi har dubbla implementeringar av FilterChain och HttpRequest, tills vi har definierat en gemensam arkitektur och paketstruktur. Annars kommer vi att få namnkonflikter och redundant kod eller vad tycker ni?

**Comment by eeebbaandersson** [comment]
Låter smart tycker jag!
Kanske är bra ifall vi ser till att få in basstrukturen som @eraiicphu gjort först och sedan addera varje filter separat?
Finns en aktiv PR för detta nu.

**Comment by eraiicphu** [comment]
Wild Eric joins conversation:
Grym ide! jag kikar igenom lite jag med så kanske vi kan harmonisera våra implementeringer. 
Mitt issue (23) handlade om att skapa base interfaces för filter och filterChain. som då framtida filters ska implementa. 

Vi tänkte på samma sak även när det kom till HttpRequest klass, men ser att du har färre fält (?) Jag speglade klassen HttpParseRequestLina som mergades den klassen tänker jag är den som parsar inkommande HTTP från browsern. 
HttpParser klassen parsar header.
och förberedde för body.

Så bara från lite snabb kik så verkar det som att mina klasserna:
(I)FilterChain 
är samma som din HttpFilter. 

Ser även att du gjort en HttpResponse. 
Tror detta fyller samma funktion som tidigare klass som mergades igår med, HttpResponseBuilder

File: src/main/java/org/example/server/FilterChain.java (hunk 1 above)
Comment (viktorlindell12) [reply src/main/java/org/example/server/FilterChain.java]: Du har helt rätt. Vi ska inte ha både org.example.filter.FilterChain (interface) och en separat klass org.example.server.FilterChain med samma namn.

**Comment by Ericthilen** [comment]
Created new pull request and added redirect and tests #95
